README
This is a "plugin" for the Video Disk Recorder (VDR).

TVGuide on GitLab: https://gitlab.com/kamel5/tvguide

Original Written by:           Louis Braun <louis DOT braun AT gmx DOT de>

Original Project's homepage:   http://projects.vdr-developer.org/projects/plg-tvguide 

Latest version available at:   https://gitlab.com/kamel5/tvguide

Original version available at: http://projects.vdr-developer.org/projects/plg-tvguide/files

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.
See the file COPYING for more information.

Requirements
------------

- Clone Git Repository: git clone git://projects.vdr-developer.org/vdr-plugin-tvguide.git
- VDR version >= 2.0.0
- Installed ImageMagick or GraphiksMagick for showing png/jpg Channel Logos, EPG Images 
  and menu icons
- for scaling the video picture to fit into the VDR menu window please use
  softhddevice plugin revision 87c1c7be (2013-01-01) or newer.
- Installed epgsearch Plugin for extended search & recording features.
- Installed remotetimers Plugin for creating timers on a vdr server remotely from a 
  vdr client.


Description
-----------

"TvGuide" is a highly customizable 2D EPG viewer plugin. The "Search & Recordings"
Menü provided by the red button allows to search in the EPG and manage timers,
search timers, series timers and switch timers in an convenient way.  

Installation
------------

After "normal" Plugin installation you have to care about the paths for the
used Icons, the channel logos and epg images. The following paths can be set
at startup:

-l path, --logopath=path
    Path to the logos (Default: <ResourceDirectory>/plugins/tvguide/channellogos/).

-e path, --epgimages=path
    Path to the epgimages (Default: <CacheDirectory>/plugins/tvguide/epgimages/).

-i path, --iconpath=path
    Path to the icons directory (Default: <ResourceDirectory>/plugins/tvguide/epgimages/).

<ResourceDirectory> and <CacheDirectory> is taken from your VDR configuration 
(make.config or vdr.pc).

During a "make install" the Icons are automatically copied from <PlgSourceDirectory>/icons/
to the default path. Included Theme files are also copied from <PlgSourceDirectory>/themes/
to <ConfigDirectory>/themes/

If directories for channel logos or icons are set via startup 
parameters, images are searched first in these directories. If the image
is not available, an additional search in the according default directory
will be performed.

I recommend to use channel logos from https://github.com/3PO/Senderlogos
To download them just change in the directory you want to place the logos
and do a:

git clone https://github.com/3PO/Senderlogos.git logos

An update of the logos can then be done with a "git pull" just inside this
directory.

For S2-6400 Users: Disable High Level OSD, otherwise the plugin will not be
loaded because lack of true color support

For Xine-Plugin Users: Set "Blend scaled Auto" as OSD display mode to achieve
an suitable true color OSD.

For Xineliboutput Users: Start vdr-sxfe with the --hud option enabled

Usage
-----

Remote Control Keys:

Up/Down/Left/Right:    Navigation in the EPG grid
Ok:                    Detailed EPG View of the selected grid
                       2nd Ok closes the detailed view
Red:                   Search & Recording Menu
Green / Yellow:	       Jump (default) five channels back / forward
Blue:                  Depends on configured Mode:
                       Default: Favorites Menu, Switch to channel in 
                       detailed EPG View
                       Optional: Switch to currently selected channel
                       Optional: Open detailed EPG view (OK switches
                       then to channel)

Numeric Keys: If Setup Option "Functionality of numeric Keys" is configured
to "Jump to specific channel", the numeric keys are used for channel number
input. If "Timely Jump" is configured, the keys are used as follows:

1 / 3:                 Big jump (default 3h) back / forward in time
4 / 6:                 huge jump (default 24h) back / forward in time
7 / 9:                 jump to previous / next prime time (8pm)

Exit:                  closes plugin

In Search & Recording menu the following options are available:

- Instant Record: create a timer for the currently selected program. An
  check for timer conflicts will be performed, if a timer conflict occurs,
  the involved timers can be adapted to solve the conflict. Alternatively
  reruns can be searched and used to solve timer conflicts. 
- Delete / Edit Timer: if a timer already exists for the currently selected
  broadcast, this timer can be deleted or edited.
- Timer Timeline: displays active timers per day
- Create Search Timer: create an search timer with epgsearch. The most
  important options can be defined directly in the menu, expert options
  are available via the "advanced options" button.
- Manage Search Timers: edit, delete and check results of search timers  
- Create Series Timer: create a periodical timer for a dedicated time on
  dedicated days.
- Create Switch Timer: create a switch timer for the currently selected
  program.
- Search: search with epgsearch
- Check for Timer Conflicts: check for timer conflicts with epgsearch.
- Search in Recordings: check if a recording already exists for the currently
  selected program.

The search & recordings menu is intended to present a convenient interface
to different VDR core recording features and additional epgsearch features.
The main goal is to reach better usability for "non expert users".

The navigation through the search & recording menu can be done easily with
up/down, left/right and the ok key. In scrolling lists, with left/right
(after the first/last option of the displayed list element is selected) you
jump to the next page of the list. Pressing up on the first / down on the 
last list item jumpps to the end / start of the list.

In the Plugin Setup Menu in Section "Recording Menus and Favorites" you can
choose three alternatives for the option "Folder for instant Recordings":
- "Always use root video folder": the recording will always be placed in the
  video root folder
- "Select from folder list": before creating a timer, you'll get asked in which
  directory the recording should be placed. If the epgsearch plugin is not 
  available, the entries from VDRs <VDRCONFIG>/folders.conf will be used. If
  the epgsearch plugin is available, the more sophisticated epgsearch mechanism
  will be used to identify the possible folders (folders from 
  <VDRCONFIG>/folders.conf and epgsearchfolders.conf and all already existing 
  folders in the video directory).
- "Use fixed folder": the configured folder will always be used. If this string
  is a epgsearch variable ("%variable%"), epgsearch will be used to replace 
  the variable with a concrete folder. All values defined in the epgsearch config
  file "epgsearchuservars.conf" can be used. These variables use the definitions
  from "epgsearchcats.conf". For a detailed description see the epgsearch Manual.
  With that and a with an EPG with detailed information about movies and series 
  it is possible to create sophisticated paths for timers automatically (for 
  instance with series season and episode in the filename).   

For a search timer also all folders and the defined dedicated folder can be used.

Favorites Menu: if configured, the blue key provides a favorite menu. In this
menu the current scheduling ("What's on now") and the next upcomming Shows 
("What's on next") can be listed. Via the Plugin Setup Menu up to four user 
defined times can be configured with a dedicated description (for instance 
"Primetime" for 8:15 pm). These user defined times are also shown as options in 
the favorites menu. Finally all search timers with the option "use as favorite" 
enabled are displayed in the favorites menu. By selecting this entry, all hits
for this search timer are shown.

Setup Options
-------------

* General Settings:

- Show Main Menu Entry
  Display "tvguide" in main menu
- Replace VDR Schedules Menu
  If set to "yes", the original VDR schedules menu will be replaced by tvguide
- Use appropriate nOpacity Theme: if nOpacity is configured as VDR skin, you can 
  configure TVGuide to select the choosen nOpacity theme automatically also as 
  TVGuide theme. If the appropriate theme is not avaiable, the default theme
  will be used. If set to "no", you can configure the theme independend of
  nOpacity
  - Theme: nOpacity independent theme to use
- Time to display in minutes: period of time to display in main EPG view
- Rounded Corners
  Use rounded corners for displayed boxes
- Channel Jump Mode (Keys Green / Yellow)
  select between jumping x channels back/forward or to the previous / next 
  channel group
- Keys Blue and OK
  Blue: Favorites in normal view, channel switch in det. EPG View, Ok: det. EPG View
  or 
  Blue: Channel Switch, Ok: Detailed EPG 
  or
  Blue: Detailed EPG, Ok: Channel Switch
- Close TVGuide after channel switch: If set to "no", tvguide will not be closed
  after a channel switch.
- Functionality of numeric Keys: see chapter Usage / Remote Control Keys
- Hide last Channel Group
  If set to yes, the channels of the last channel group will not be displayed
- Big Step (Keys 1 / 3) in hours
  Hours to jump vertically with keys 1 / 3
- Huge Step (Keys 4 / 6) in hours
  Hours to jump vertically with keys 4 / 6
- Time Format (12h/24h)
- Display Reruns in detailed EPG View

* Screen Presentation:

- Display Mode
  Choose between horizontal or vertical display of the channels. All further
  width / height settings are dependend of this setting
- Width/Height of Channel Header (% of osd width/height)
- Width/Height of Timeline (% of osd width/height)
- Number of Channels to display
- Height of Headers (Status Header and EPG View Header, % of osd height)
- Height of Footer
- Display status header
  - Scale video to upper right corner
  - Rounded corners around video frame
- Display channel names in Header
- Display channel groups
  - Width/Height of channel groups (% of osd width/height)
- Display current time baseline
- Show Channel Logos
  show / hide channel logos, if logos are shown:
  - Logo Extension
    jpg / png
  - Logo width ratio
  - Logo height ratio
- Text Border in Detailed EPG View (in pixel)
- Show EPG Images
  show / hide EPG images, if images are shown:
  - EPG Image width ratio
  - EPG Image height ratio
  - Number of additional EPG Images: number of additional EPG
    images displayed at the bottom of the detailed EPG view.
  - Additional EPG Image width (in Pixel)
  - Additional EPG Image height (in Pixel)
	
* Fonts and Fontsizes:

- Font:
  Used Font, all Fonts installed on your system are shown
- adjust various font sizes:
  the defined value is added to / subtracted from the default font size

* Recording Menus and Favorites

- Folder for instant Recordings: (see chapter "Usage" for detailed description)
  - Always use root video folder
  - Select from folder list
  - Use fixed folder
- Use Remotetimers: Timers are handled not locally but on a VDR Server defined
  with the remotetimers plugin.
- Use "What's on now" in favorites
- Use "What's on next" in favorites
- Use user defined time 1 (to 4) in favorites
  - Description (1 - 4)
  - Time (1 - 4)
- Limit channels in favorites: if set to yes, only the configured channel range
  is used to display favorite results (except for search timer favorites)
  - start channel
  - stop channel

* Image Caching

- Create Log Messages for image loading: enable debug messages for image loading
- Limit Logo Cache: if set to "no", every channel logo is cached. if set to "yes",
  only the amount configured for "Maximal number of logos to cache" is cached, all
  other logos are always loaded "on the fly"
  - Maximal number of logos to cache
- Number of  logos to cache at start: number of logos to pre cache at plugin startup
  (a too large number causes a slower VDR start)
- Information about image cache sizes


HISTORY
VDR Plugin 'tvguide' Revision History
-------------------------------------

2012-08-12: Version 0.0.1

- Initial revision.

2013-01-17: Version 0.0.2

- Changed color buttons to nOpacity style
- Changed channelheader to display transparent logos properly
- Added "style nOpacity" for backgrounds and theme nOpacity

2013-05-13: Version 0.0.3

- Added dummy grids if no EPG information is available for a certain
  time
- Completely rewrote code for creation of epg grids for a channel column
- Completely rewrote code for scrolling forward and backward in time
- Added status header with epg information of currently selected 
  schedule and scaled video picture

2013-05-24: Version 0.0.4

- Introduced horizontal display

2013-06-01: Version 0.0.5

- Added Theme DarkRed
- Fixed some Bugs (position of video, deadlock)
- Added Theme IceBlue
- setup of usage of blending now done with clrDoBlending theme color
  variable and not by setup
- font color of active grid themable
- avoided nasty font pixelation effects with theme iceblue
- Display of channel groups
- Buttons green / yellow can be configured to jump to prev / next channel
  group
- Added setup option to hide last channel group
- Display of Main Menu Entry configurable
- Eliminated setup option "Number of channels to jump", directly jump 
  number of channel columns / rows with green / yellow
- Added setup option to switch functionality of keys "Blue" and "OK"
- Setup option to hide schedules time display in horizontal EPG grids

2013-07-08: Version 0.0.6

- added frame around scaled video picture
- added theme "keep it simple" (thanks @saman)
- display of additional EPG pictures in detailed epg view
- Introduction of "Search & Recording" Menu
- added possibility to replace original VDR schedules menu
- changed channel jump logic
- red button also available in detail view
- page scrolling with keys left and right in detail view
- added possibility to choose folder for instant recording, possible
  folders are read from VDRs folders list
- recording folder also choosable for series timers
- blue button also available in detail view
- color buttons displayed as configured in VDR
- Added Setup Option to define if tvguide closes after channel switching
- Added remotetimers support (thanks @Saman for providing a patch)
- Pimped timeline

2013-07-23: Version 1.0.0

- added tvscraper support

Version 1.1.0

- changed build system
- introduced image caching
- introduced new "graphical" style
- added new graphical default theme and theme darkred NG corresponding 
  to nOpacity themes
- using automatically configured nOpacity theme if available
- adapted startup options to nOpacity startup options:
  -e epgimages directory
  -i icons directory
  -l logo directory
- changed detailed epg view using full screen, some further optimisations
- search for <event_id>_0.jpg beside <event_id>.jpg as epg image
- implemented GraphicsMagick compatibility (thanks @Manuel Reimer
  for providing the patch)
- changed Makefile to support both ImageMagick and GraphicsMagick
  (configurable in Makefile)
- Added russian and catalan translation
- Added slovakian translation
- fixed channel switching with blue key if detailed epg view is opened and
  if "close tvguide on channel switch" is configured
- fixed wrong font for clock in horizontal view
- Added feature to jump to a specific channel with number keys
- Fixed Bug 1484
- Added Event Short Text (if available) to RecName (Closes Ticket 1490)
- Fixed OSD Background Color (Closes Ticket 1474)
- Rounded Corners for color buttons (Closes Ticket 1475)
- Fixed some issues with text backgrounds in flat themes (closes 
  Tickets 1480 and 1486)
- Added possibility to search for reruns in case of a timer conflict
- Added timer Timeline view in red button recording menu
- Current channel positioned in middle of channellist when tvguide is 
  opened
- Added possibility to manage EPG Search Timers in red button recording 
  menu
- completely refactored recording menu
- restructured detailöed EPG View that tv frame and status buttons 
  are displayed
- Favorites menu with EPG Search favorite timers with blue key
- Display baseline for current time
- Added possibility to create timer from favorites results
- left / right scroll page up / down in recmenu result lists
- Improved favorites menu with whats on now / next and up to four user
  defined times
- also check default paths when images and icons are loaded
- directory for timers can be edited in timer edit menu
- series recording can be configured in searchtimer menus
- directory for searchtimers can be set and edited in searchtimer menus
- set blue key favorite mode as new default

Version 1.2.0

- changed video directory choosage behaviour: added possibility to use 
  a fixed recording dir which also supports epgsearch variables
- fixed a bug when scrolling page up and the menu footer is active
- order of search timers in search timer list in alphabetical order
- changed order of actions on search timer menu items

Version 1.2.1

- changed event result lists that it is possible to scroll page up and
  down directly with left / right. A recording is triggered now with the
  red key.
- Added possibility to create a recording from search timer result list
- Set minimum time to display in Plugin Setup from 120 to 60 minutes
- made order of search timer list case insensitive

Version 1.2.2

- added scraper2vdr support
- introduced tabbed EPG detail view

Version 1.2.3

- Compile under VDR 2.3.1+
- Reinsert old tvscraper compatibility
- New Makefile style
- Show inactive timer in EPG grid
- Center channel logos in EPG
- Clock in timeline if displaymode horizontal

Version 1.2.4

- Add Support for new SVDRPPeering
- Add italian translation
- Use graphicsmagick instead of imagemagick
- Correct a switchtimer error
- Add "switchMinsBefore" to setup menu
- Eliminate some warnings
- Center channel logos in favorite view
- Update "services/epgsearch.h"

Version 1.2.5


config.c
#include <string>
#include "config.h"

cTvguideConfig::cTvguideConfig() {
    debugImageLoading = 0;
    showMainMenuEntry = 1;
    replaceOriginalSchedule = 0;
    displayMode = eHorizontal;
    showTimeInGrid = 1;
    channelCols = 5;
    channelRows = 10;
    displayTime = 160;
    displayHorizontalTime = 160;
    displayStatusHeader = 1;
    displayChannelGroups = 1;
    displayTimeBase = 1;
    headerHeightPercent = 20;
    channelGroupsPercent = 5;
    epgViewBorder = 50;
    scaleVideo = 1;
    decorateVideo = 1;
    timeLineWidthPercent = 8;
    timeLineHeightPercent = 4;
    displayChannelName = 1;
    channelHeaderWidthPercent = 20;
    channelHeaderHeightPercent = 15;
    footerHeightPercent = 7;
    stepMinutes = 30;
    bigStepHours = 3;
    hugeStepHours = 24;
    channelJumpMode = eNumJump;
    jumpChannels = 0;
    blueKeyMode = 2;
    closeOnSwitch = 1;
    numkeyMode = 0;
    useRemoteTimers = 0;
    hideLastGroup = 0;
    hideChannelLogos = 0;
    logoWidthRatio = 13;
    logoHeightRatio = 10;
    logoExtension = 0;
    hideEpgImages = 0;
    epgImageWidth = 315;
    epgImageHeight = 240;
    numAdditionalEPGPictures = 9;
    epgImageWidthLarge = 525;
    epgImageHeightLarge = 400;
    detailedViewScrollStep = 5;
    instRecFolderMode = eFolderRoot;
    instRecFixedFolder = "";
    favWhatsOnNow = 1;
    favWhatsOnNext = 1;
    favUseTime1 = 0;
    favUseTime2 = 0;
    favUseTime3 = 0;
    favUseTime4 = 0;
    favTime1 = 0;
    favTime2 = 0;
    favTime3 = 0;
    favTime4 = 0;
    descUser1 = "";
    descUser2 = "";
    descUser3 = "";
    descUser4 = "";
    favLimitChannels = 0;
    favStartChannel = 0;
    favStopChannel = 0;
    switchMode = 0;
    switchMinsBefore = 2;
    fontIndex = 0;
    fontNameDefault = "VDRSymbols Sans:Book";
    FontButtonDelta = 0;
    FontDetailViewDelta = 0;
    FontDetailViewSmallDelta = 0;
    FontDetailHeaderDelta = 0;
    FontMessageBoxDelta = 0;
    FontMessageBoxLargeDelta = 0;
    FontStatusHeaderDelta = 0;
    FontStatusHeaderLargeDelta = 0;
    FontChannelHeaderDelta = 0;
    FontChannelGroupsDelta = 0;
    FontGridDelta = 0;
    FontGridSmallDelta = 0;
    FontTimeLineWeekdayDelta = 0;
    FontTimeLineDateDelta = 0;
    FontTimeLineTimeDelta = 0;
    FontChannelHeaderHorizontalDelta = 0;
    FontChannelGroupsHorizontalDelta = 0;
    FontGridHorizontalDelta = 0;
    FontGridHorizontalSmallDelta = 0;
    FontTimeLineDateHorizontalDelta = 0;
    FontTimeLineTimeHorizontalDelta = 0;
    FontRecMenuItemDelta = 0;
    FontRecMenuItemSmallDelta = 0;
    FontRecMenuItemLargeDelta = 0;
    timeFormat = 1;
    useNopacityTheme = 1;
    themeIndex = -1;
    themeIndexCurrent = -1;
    themeName = "";
    nOpacityTheme = "";
    style = eStyleBlendingDefault;
    roundedCorners = 0;
    displayRerunsDetailEPGView = 1;
    numReruns = 5;
    useSubtitleRerun = 1;
    numLogosInitial = 30;
    numLogosMax = 50;
    limitLogoCache = 1;
    logoPathSet = false;
    imagesPathSet = false;
    iconsPathSet = false;
}

cTvguideConfig::~cTvguideConfig() {
}

void cTvguideConfig::setDynamicValues() {
    numGrids = (displayMode == eVertical)?channelCols:channelRows;
    jumpChannels = numGrids;
}

bool cTvguideConfig::LoadTheme() {
    //is correct theme already loaded?
    if (nOpacityTheme.size() == 0)
        nOpacityTheme = Setup.OSDTheme;
    if ((themeIndex > -1) && (themeIndex == themeIndexCurrent)) {
        if (!nOpacityTheme.compare(Setup.OSDTheme)) {
            return false;
        } else {
            nOpacityTheme = Setup.OSDTheme;
        }
    }
    //Load available Themes
    cThemes themes;
    themes.Load(*cString("tvguide"));
    int numThemesAvailable = themes.NumThemes();

    //Check for nOpacity Theme
    if (useNopacityTheme) {
        std::string nOpacity = "nOpacity";
        std::string currentSkin = Setup.OSDSkin;
        std::string currentSkinTheme = Setup.OSDTheme;
        if (!currentSkin.compare(nOpacity)) {
            for (int curThemeIndex = 0; curThemeIndex < numThemesAvailable; curThemeIndex++) {
                std::string curThemeName = themes.Name(curThemeIndex);
                if (!curThemeName.compare(currentSkinTheme)) {
                    themeIndex = curThemeIndex;
                    break;
                }
            }
        }
    }

    if (themeIndex == -1) {
        for (int curThemeIndex = 0; curThemeIndex < numThemesAvailable; curThemeIndex++) {
            std::string curThemeName = themes.Name(curThemeIndex);
            if (!curThemeName.compare("default")) {
                themeIndex = curThemeIndex;
                break;
            }
        }

    }

    if (themeIndex == -1)
        themeIndex = 0;

    themeIndexCurrent = themeIndex;

    const char *themePath = themes.FileName(themeIndex);
    if (access(themePath, F_OK) == 0) {
        ::theme.Load(themePath);
        themeName = themes.Name(themeIndex);        
    }
    esyslog("tvguide: set Theme to %s", *themeName);
    return true;
}


void cTvguideConfig::SetStyle(void) {
    if (theme.Color(clrStyle) == CLR_STYLE_FLAT) {
        style = eStyleFlat;
        esyslog("tvguide: set flat style");
    } else if (theme.Color(clrStyle) == CLR_STYLE_BLENDING_DEFAULT) {
        style = eStyleBlendingDefault;
        esyslog("tvguide: set blending style");
    } else if (theme.Color(clrStyle) == CLR_STYLE_GRAPHICAL) {
        style = eStyleGraphical;
        esyslog("tvguide: set graphical style");
    } else {
        style = eStyleBlendingMagick;
        esyslog("tvguide: set magick blending style");
    }
    
}

void cTvguideConfig::SetLogoPath(cString path) {
    logoPath = checkSlashAtEnd(*path);
    logoPathSet = true;
    esyslog("tvguide: Logo Path set to %s", *logoPath);
}

void cTvguideConfig::SetImagesPath(cString path) {
    epgImagePath = checkSlashAtEnd(*path);
    imagesPathSet = true;
    esyslog("tvguide: EPG Image Path set to %s", *epgImagePath);
}

void cTvguideConfig::SetIconsPath(cString path) {
    iconPath = checkSlashAtEnd(*path);
    iconsPathSet = true;
    esyslog("tvguide: Icon Path set to %s", *iconPath);
}

void cTvguideConfig::SetDefaultPathes(void) {
    logoPathDefault = cString::sprintf("%s/logos/", cPlugin::ResourceDirectory(PLUGIN_NAME_I18N));
    iconPathDefault = cString::sprintf("%s/icons/", cPlugin::ResourceDirectory(PLUGIN_NAME_I18N));
    epgImagePathDefault = cString::sprintf("%s/epgimages/", cPlugin::CacheDirectory(PLUGIN_NAME_I18N));

    if (!logoPathSet) {
        logoPath = logoPathDefault;
    }
    if (!imagesPathSet) {
        epgImagePath = epgImagePathDefault;
    }
    if (!iconsPathSet) {
        iconPath = iconPathDefault;
    }
}

cString cTvguideConfig::checkSlashAtEnd(std::string path) {
    try {
        if (!(path.at(path.size()-1) == '/'))
            return cString::sprintf("%s/", path.c_str());
    } catch (...) {return path.c_str();}
    return path.c_str();
}

bool cTvguideConfig::SetupParse(const char *Name, const char *Value) {
    if      (strcmp(Name, "timeFormat") == 0)               timeFormat = atoi(Value);
    else if (strcmp(Name, "debugImageLoading") == 0)        debugImageLoading = atoi(Value);
    else if (strcmp(Name, "showMainMenuEntry") == 0)        showMainMenuEntry = atoi(Value);
    else if (strcmp(Name, "replaceOriginalSchedule") == 0)  replaceOriginalSchedule = atoi(Value);
    else if (strcmp(Name, "useNopacityTheme") == 0)         useNopacityTheme = atoi(Value);   
    else if (strcmp(Name, "themeIndex") == 0)               themeIndex = atoi(Value);
    else if (strcmp(Name, "displayMode") == 0)              displayMode = atoi(Value);
    else if (strcmp(Name, "showTimeInGrid") == 0)           showTimeInGrid = atoi(Value);
    else if (strcmp(Name, "displayStatusHeader") == 0)      displayStatusHeader = atoi(Value);
    else if (strcmp(Name, "displayChannelGroups") == 0)     displayChannelGroups = atoi(Value);
    else if (strcmp(Name, "displayTimeBase") == 0)          displayTimeBase = atoi(Value);
    else if (strcmp(Name, "headerHeightPercent") == 0)      headerHeightPercent = atoi(Value);
    else if (strcmp(Name, "channelGroupsPercent") == 0)     channelGroupsPercent = atoi(Value);
    else if (strcmp(Name, "epgViewBorder") == 0)            epgViewBorder = atoi(Value);
    else if (strcmp(Name, "scaleVideo") == 0)               scaleVideo = atoi(Value);
    else if (strcmp(Name, "decorateVideo") == 0)            decorateVideo = atoi(Value);
    else if (strcmp(Name, "roundedCorners") == 0)           roundedCorners = atoi(Value);
    else if (strcmp(Name, "channelCols") == 0)              channelCols = atoi(Value);
    else if (strcmp(Name, "channelRows") == 0)              channelRows = atoi(Value);
    else if (strcmp(Name, "displayTime") == 0)              displayTime = atoi(Value);
    else if (strcmp(Name, "displayHorizontalTime") == 0)    displayHorizontalTime = atoi(Value);
    else if (strcmp(Name, "hideChannelLogos") == 0)         hideChannelLogos = atoi(Value);
    else if (strcmp(Name, "logoExtension") == 0)            logoExtension = atoi(Value);
    else if (strcmp(Name, "logoWidthRatio") == 0)           logoWidthRatio = atoi(Value);
    else if (strcmp(Name, "logoHeightRatio") == 0)          logoHeightRatio = atoi(Value);
    else if (strcmp(Name, "bigStepHours") == 0)             bigStepHours = atoi(Value);
    else if (strcmp(Name, "hugeStepHours") == 0)            hugeStepHours = atoi(Value);
    else if (strcmp(Name, "channelJumpMode") == 0)          channelJumpMode = atoi(Value);
    else if (strcmp(Name, "blueKeyMode") == 0)              blueKeyMode = atoi(Value);
    else if (strcmp(Name, "numkeyMode") == 0)               numkeyMode = atoi(Value);
    else if (strcmp(Name, "closeOnSwitch") == 0)            closeOnSwitch = atoi(Value);
    else if (strcmp(Name, "useRemoteTimers") == 0)          useRemoteTimers = atoi(Value);
    else if (strcmp(Name, "hideLastGroup") == 0)            hideLastGroup = atoi(Value);
    else if (strcmp(Name, "hideEpgImages") == 0)            hideEpgImages = atoi(Value);
    else if (strcmp(Name, "epgImageWidth") == 0)            epgImageWidth = atoi(Value);
    else if (strcmp(Name, "epgImageHeight") == 0)           epgImageHeight = atoi(Value);
    else if (strcmp(Name, "numAdditionalEPGPictures") == 0) numAdditionalEPGPictures = atoi(Value);
    else if (strcmp(Name, "epgImageWidthLarge") == 0)       epgImageWidthLarge = atoi(Value);
    else if (strcmp(Name, "epgImageHeightLarge") == 0)      epgImageHeightLarge = atoi(Value);
    else if (strcmp(Name, "detailedViewScrollStep") == 0)   detailedViewScrollStep = atoi(Value);
    else if (strcmp(Name, "timeLineWidthPercent") == 0)     timeLineWidthPercent = atoi(Value);
    else if (strcmp(Name, "timeLineHeightPercent") == 0)    timeLineHeightPercent = atoi(Value);
    else if (strcmp(Name, "displayChannelName") == 0)       displayChannelName = atoi(Value);
    else if (strcmp(Name, "channelHeaderWidthPercent") == 0)  channelHeaderWidthPercent = atoi(Value);
    else if (strcmp(Name, "channelHeaderHeightPercent") == 0) channelHeaderHeightPercent = atoi(Value);
    else if (strcmp(Name, "footerHeightPercent") == 0)      footerHeightPercent = atoi(Value);
    else if (strcmp(Name, "instRecFolderMode") == 0)        instRecFolderMode = atoi(Value);
    else if (strcmp(Name, "instRecFixedFolder") == 0)       instRecFixedFolder = Value;
    else if (strcmp(Name, "favWhatsOnNow") == 0)            favWhatsOnNow = atoi(Value);
    else if (strcmp(Name, "favWhatsOnNext") == 0)           favWhatsOnNext = atoi(Value);
    else if (strcmp(Name, "favUseTime1") == 0)              favUseTime1 = atoi(Value);
    else if (strcmp(Name, "favUseTime2") == 0)              favUseTime2 = atoi(Value);
    else if (strcmp(Name, "favUseTime3") == 0)              favUseTime3 = atoi(Value);
    else if (strcmp(Name, "favUseTime4") == 0)              favUseTime4 = atoi(Value);
    else if (strcmp(Name, "favTime1") == 0)                 favTime1 = atoi(Value);
    else if (strcmp(Name, "favTime2") == 0)                 favTime2 = atoi(Value);
    else if (strcmp(Name, "favTime3") == 0)                 favTime3 = atoi(Value);
    else if (strcmp(Name, "favTime4") == 0)                 favTime4 = atoi(Value);
    else if (strcmp(Name, "descUser1") == 0)                descUser1 = Value;
    else if (strcmp(Name, "descUser2") == 0)                descUser2 = Value;
    else if (strcmp(Name, "descUser3") == 0)                descUser3 = Value;
    else if (strcmp(Name, "descUser4") == 0)                descUser4 = Value;
    else if (strcmp(Name, "favLimitChannels") == 0)         favLimitChannels = atoi(Value);
    else if (strcmp(Name, "favStartChannel") == 0)          favStartChannel = atoi(Value);
    else if (strcmp(Name, "favStopChannel") == 0)           favStopChannel = atoi(Value);
    else if (strcmp(Name, "switchMode") == 0)               switchMode = atoi(Value);
    else if (strcmp(Name, "switchMinsBefore") == 0)         switchMinsBefore = atoi(Value);
    else if (strcmp(Name, "fontIndex") == 0)                fontIndex = atoi(Value);
    else if (strcmp(Name, "FontButtonDelta") == 0)          FontButtonDelta = atoi(Value);
    else if (strcmp(Name, "FontDetailViewDelta") == 0)      FontDetailViewDelta = atoi(Value);
    else if (strcmp(Name, "FontDetailHeaderDelta") == 0)    FontDetailHeaderDelta = atoi(Value);
    else if (strcmp(Name, "FontMessageBoxDelta") == 0)      FontMessageBoxDelta = atoi(Value);  
    else if (strcmp(Name, "FontMessageBoxLargeDelta") == 0) FontMessageBoxLargeDelta = atoi(Value); 
    else if (strcmp(Name, "FontStatusHeaderDelta") == 0)    FontStatusHeaderDelta = atoi(Value);    
    else if (strcmp(Name, "FontStatusHeaderLargeDelta") == 0) FontStatusHeaderLargeDelta = atoi(Value);
    else if (strcmp(Name, "FontChannelHeaderDelta") == 0)   FontChannelHeaderDelta = atoi(Value);
    else if (strcmp(Name, "FontChannelGroupsDelta") == 0)   FontChannelGroupsDelta = atoi(Value);
    else if (strcmp(Name, "FontGridDelta") == 0)            FontGridDelta = atoi(Value);    
    else if (strcmp(Name, "FontGridSmallDelta") == 0)       FontGridSmallDelta = atoi(Value);   
    else if (strcmp(Name, "FontTimeLineWeekdayDelta") == 0) FontTimeLineWeekdayDelta = atoi(Value); 
    else if (strcmp(Name, "FontTimeLineDateDelta") == 0)    FontTimeLineDateDelta = atoi(Value);    
    else if (strcmp(Name, "FontTimeLineTimeDelta") == 0)    FontTimeLineTimeDelta = atoi(Value);
    else if (strcmp(Name, "FontChannelHeaderHorizontalDelta") == 0) FontChannelHeaderHorizontalDelta = atoi(Value);
    else if (strcmp(Name, "FontChannelGroupsHorizontalDelta") == 0) FontChannelGroupsHorizontalDelta = atoi(Value);
    else if (strcmp(Name, "FontGridHorizontalDelta") == 0)  FontGridHorizontalDelta = atoi(Value);
    else if (strcmp(Name, "FontGridHorizontalSmallDelta") == 0) FontGridHorizontalSmallDelta = atoi(Value);
    else if (strcmp(Name, "FontTimeLineDateHorizontalDelta") == 0) FontTimeLineDateHorizontalDelta = atoi(Value);
    else if (strcmp(Name, "FontTimeLineTimeHorizontalDelta") == 0) FontTimeLineTimeHorizontalDelta = atoi(Value);
    else if (strcmp(Name, "FontRecMenuItemDelta") == 0)     FontRecMenuItemDelta = atoi(Value);
    else if (strcmp(Name, "FontRecMenuItemSmallDelta") == 0) FontRecMenuItemSmallDelta = atoi(Value);
    else if (strcmp(Name, "FontRecMenuItemLargeDelta") == 0) FontRecMenuItemLargeDelta = atoi(Value);
    else if (strcmp(Name, "displayRerunsDetailEPGView") == 0) displayRerunsDetailEPGView = atoi(Value);
    else if (strcmp(Name, "numReruns") == 0)                numReruns = atoi(Value);
    else if (strcmp(Name, "useSubtitleRerun") == 0)         useSubtitleRerun = atoi(Value);
    else if (strcmp(Name, "numLogosInitial") == 0)         numLogosInitial = atoi(Value);
    else if (strcmp(Name, "numLogosMax") == 0)             numLogosMax = atoi(Value);
    else if (strcmp(Name, "limitLogoCache") == 0)          limitLogoCache = atoi(Value);
    else return false;
    return true;
}
